conan/tools/google/bazeldeps2.py

```diff
@@ -0,0 +1,519 @@
+def _relativize_path(path, relative_to_path):
+    """
+    Relativize the path with regard to a given base path
+
+    :param path: path to relativize
+    :param relative_to_path: base path to prune from the path
+    :return: Unix-like path relative to ``relative_to_path``. Otherwise, it returns
+             the Unix-like ``path``.
+    """
+    if not path or not relative_to_path:
+        return path
+    p = path.replace("\\", "/")
+    rel = relative_to_path.replace("\\", "/")
+    if p.startswith(rel):
+        return p[len(rel):].lstrip("/")
+    elif rel in p:
+        return p.split(rel)[-1].lstrip("/")
+    else:
+        return p.lstrip("/")
+
+
+class _BazelDependenciesBZLGenerator:
+
+    filename = "dependencies.bzl"
+    template = textwrap.dedent("""\
+        # This Bazel module should be loaded by your WORKSPACE file.
+        # Add these lines to your WORKSPACE one (assuming that you're using the "bazel_layout"):
+        # load("@//bazel-conan-tools:dependencies.bzl", "load_conan_dependencies")
+        # load_conan_dependencies()
+
+        {%- macro new_local_repository(pkg_name, pkg_folder, pkg_build_file_path) %}
+            native.new_local_repository(
+                name="{{pkg_name}}",
+                path="{{pkg_folder}}",
+                build_file="{{pkg_build_file_path}}",
+            )
+        {%- endmacro %}
+
+        def load_conan_dependencies():
+            {%- for pkg_name, pkg_folder, pkg_build_file_path in dependencies %}
+            {{new_local_repository(pkg_name, pkg_folder, pkg_build_file_path)}}
+            {%- endfor %}
+        """)
+
+    def __init__(self, conanfile, dependencies):
+        self._conanfile = conanfile
+        # Containing the [(pkg_name, pkg_folder, pkg_build_file_path), ...]
+        self._dependencies = dependencies
+
+    def generate(self):
+        template = Template(self.template, trim_blocks=True, lstrip_blocks=True,
+                            undefined=StrictUndefined)
+        content = template.render(self._dependencies)
+        # Saving the BUILD (empty) and dependencies.bzl files
+        save(self.filename, content)
+        save("BUILD.bazel", "# This is an empty BUILD file to be able to load the "
+                            "dependencies.bzl one.")
+
+
+class _BazelBUILDGenerator:
+
+    filename = "BUILD.bazel"
+    template = textwrap.dedent("""
+    load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")
+
+    {% for libname, filepath in libs.items() %}
+    cc_import(
+        name = "{{ libname }}_precompiled",
+        {{ library_type }} = "{{ filepath }}",
+    )
+    {% endfor %}
+
+    {% for libname, (lib_path, dll_path) in shared_with_interface_libs.items() %}
+    cc_import(
+        name = "{{ libname }}_precompiled",
+        interface_library = "{{ lib_path }}",
+        shared_library = "{{ dll_path }}",
+    )
+    {% endfor %}
+
+    cc_library(
+        name = "{{ name }}",
+        {% if headers %}
```

Comment:
```suggestion
        {%- if headers %}
```
Doing this everywhere prevents the lines of the generated file from being spread.